src/utils/loadFileHandler.ts

```diff
@@ -145,19 +151,188 @@ const loadData = (buffer: BufferReader) =>
     new Float32Array(new Uint8Array(buffer!.readChunk()).buffer);
 
 const loadBits = (buffer: BufferReader) =>
-    buffer!.hasBitsData()
-        ? null
-        : new Uint16Array(new Uint8Array(buffer!.readChunk()).buffer);
+    new Uint16Array(new Uint8Array(buffer!.readChunk()).buffer);
+
+const loadPPK2File = async (
+    filename: string,
+    onProgress: (message: string, percentage: number) => void
+) => {
+    let progress = 0;
+    let lastUpdate = 0;
+
+    const outputPath = path.join(os.tmpdir(), v4());
+    fs.mkdirSync(outputPath);
+    const cleanUp = () => {
+        fs.rmSync(outputPath, { recursive: true, force: true });
+    };
+    window.addEventListener('beforeunload', cleanUp);
```

Comment:
not sure classes / utils are the good place to add window events as the side effect. might be better to do that as a part of the react life cycle?

Reply:
This is to save guard that if window is closed before the release() is called then object cleans it self. This was inspired  from this [model](https://en.cppreference.com/w/cpp/language/raii)